src/catchup/simulation/ApplyTransactionsWork.cpp

```diff
@@ -35,17 +49,190 @@ ApplyTransactionsWork::ApplyTransactionsWork(
     }
 }
 
+static void
+checkResults(Application& app, uint32_t ledger,
+             std::vector<TransactionResultPair> const& results)
+{
+    auto resSet = TransactionFrame::getTransactionHistoryResults(
+        app.getDatabase(), ledger);
+
+    assert(resSet.results.size() == results.size());
+    for (size_t i = 0; i < results.size(); i++)
+    {
+        auto const& res = results[i];
+        assert(res.transactionHash == resSet.results[i].transactionHash);
+
+        auto const& dbRes = resSet.results[i].result;
+        if (dbRes.result.code() != res.result.result.code())
+        {
+            auto msg = fmt::format(
+                "Expected result code {} does not agree with {} for tx {}",
+                res.result.result.code(), dbRes.result.code(),
+                binToHex(res.transactionHash));
+            CLOG(ERROR, "History") << msg;
+        }
+    }
+}
+
+bool
+ApplyTransactionsWork::hasSig(PublicKey const& account,
+                              TransactionEnvelope const& oldEnvelope,
+                              Hash const& hash)
+{
+    // Is the signature of this account present in the envelope we're
+    // simulating?
+    if (!mUsedPubKeys.insert(account).second)
+    {
+        return false;
+    }
+
+    for (auto const& sig : oldEnvelope.signatures)
+    {
+        if (PubKeyUtils::verifySig(account, sig.signature, hash))
```

Comment:
checking signatures like this bypasses using the "hint" - see `SignatureUtils::verify`